### DISCUSSION TITLE
The Bypass and Preview Cookie Names should be configurable

### DISCUSSION THREAD

**Discussion by ashleyfrieze** [vercel/next.js#36729] [Ideas]
### Describe the feature you'd like to request

At the moment, the cookies used for bypass and preview data are hardcoded:

```js
// next.js
export const COOKIE_NAME_PRERENDER_BYPASS = `__prerender_bypass`
export const COOKIE_NAME_PRERENDER_DATA = `__next_preview_data`
```

This lends itself to a DDOS risk where someone can attempt to introduce these cookies to their requests in an attempt to bypass any edge-level caching.

### Describe the solution you'd like

If these could be configurable, perhaps via server-side environment variables, then we could add obscurity to the security of these cookies.

### Describe alternatives you've considered

An alternative might be introducing some custom middleware to rewrite the cookies for each request/response. If there's a recipe for doing that it might be useful.

That said, this is an easy to use vulnerability that could easily be closed by the use of environment variables and a recommendation in the docs to introduce a custom cookie name - perhaps with an environment-specific hash in it.

**Comment by cvanh** [comment]
I can get behind the story that @ashleyfrieze is pitching. 

An other argument could be that these cookies are prefixed with `_` which in a lot of cases the caching server strips from the request because this is often used by trackers such as: [hotjar](https://help.hotjar.com/hc/en-us/articles/6952777582999-Cookies-Set-by-the-Hotjar-Tracking-Code), google analytics, [meta pixel](https://developers.facebook.com/docs/marketing-api/conversions-api/parameters/fbp-and-fbc/), [matomo](https://matomo.org/faq/how-to/how-do-i-enforce-tracking-without-cookies/), [livechat](https://www.livechat.com/legal/privacy-policy/) just to name a few.

If we were able to change the prerender cookie people can more easily improve their cache hit rate because they can block all cookies starting `_`
